test/integration/client-side-encryption/range_explicit_encryption.prose.test.ts

```diff
@@ -32,21 +32,21 @@ function byId(a, b) {
   return 0;
 }
 
-function compareNumericValues(
-  type: typeof dataTypes[number]['type'],
-  value: unknown,
-  expectedValue: number
-) {
+function compareNumericValuesFactory(
+  type: typeof dataTypes[number]['type']
+): (value: unknown, expected: number) => void {
   if (type === 'Date') {
-    expect(value).to.be.instanceof(Date);
-    expect((value as Date).getUTCMilliseconds()).to.equal(expectedValue);
+    return (value, expected) => {
+      expect(value).to.be.instanceof(Date);
+      expect((value as Date).getUTCMilliseconds()).to.equal(expected);
+    };
   } else if (type.includes('Decimal')) {
-    expect(value).to.be.instanceof(Decimal128);
-    expect((value as Decimal128).bytes).to.deep.equal(
-      new Decimal128(expectedValue.toString()).bytes
-    );
+    return (value, expected) => {
+      expect(value).to.be.instanceof(Decimal128);
+      expect((value as Decimal128).bytes).to.deep.equal(new Decimal128(expected.toString()).bytes);
+    };
   } else {
-    expect(value).to.equal(expectedValue);
+    return (value, expected) => expect(value).to.equal(expected);
   }
 }
```

Comment:
Can we use chai's expect().deep.equal instead, in all cases, so we shouldn't need a helper? unless I am missing something

Reply:
see my response to your other comment